catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/converter/DorisExceptionConverter.java

```diff
@@ -6,15 +6,70 @@
 
 import com.datastrato.gravitino.catalog.jdbc.converter.JdbcExceptionConverter;
 import com.datastrato.gravitino.exceptions.GravitinoRuntimeException;
+import com.datastrato.gravitino.exceptions.NoSuchColumnException;
+import com.datastrato.gravitino.exceptions.NoSuchSchemaException;
+import com.datastrato.gravitino.exceptions.NoSuchTableException;
+import com.datastrato.gravitino.exceptions.SchemaAlreadyExistsException;
+import com.datastrato.gravitino.exceptions.TableAlreadyExistsException;
+import com.datastrato.gravitino.exceptions.UnauthorizedException;
+import com.google.common.annotations.VisibleForTesting;
 import java.sql.SQLException;
+import java.util.regex.Pattern;
 
 /** Exception converter to Gravitino exception for Doris. */
 public class DorisExceptionConverter extends JdbcExceptionConverter {
+
+  // see: https://doris.apache.org/docs/admin-manual/maint-monitor/doris-error-code/
+  @VisibleForTesting static final int CODE_DATABASE_EXISTS = 1007;
+
+  static final int CODE_TABLE_EXISTS = 1050;
+  static final int CODE_NO_SUCH_SCHEMA = 1049;
+  static final int CODE_DATABASE_NOT_EXISTS = 1008;
+  static final int CODE_NO_SUCH_TABLE = 1051;
+  static final int CODE_UNAUTHORIZED = 1045;
+  static final int CODE_NO_SUCH_COLUMN = 1054;
+  static final int CODE_OTHER = 1105;
+
+  private static final String DATABASE_ALREADY_EXISTS_PATTERN_STRING =
+      ".*detailMessage = Can't create database '.*'; database exists";
+  private static final Pattern DATABASE_ALREADY_EXISTS_PATTERN =
+      Pattern.compile(DATABASE_ALREADY_EXISTS_PATTERN_STRING);
+
   @SuppressWarnings("FormatStringAnnotation")
   @Override
   public GravitinoRuntimeException toGravitinoException(SQLException se) {
-    // TODO: add implementation for doris catalog
+    int errorCode = se.getErrorCode();
```

Comment:
cc @xloya For relational storage backend, could we add the similar logic?

Reply:
I think it's OK, but we may need to add the database type in the error code processing logic, because the error codes defined by different databases may be different for one error.

Reply:
Maybe you can define a similar interface.